### ISSUE TITLE
Server error when trying to refund an order for more than the balance of the order

### ISSUE THREAD

**Issue by kieckhafer** [reactioncommerce/reaction#3704]
### Expected behavior
When trying to refund an amount that's greater than the balance of the order, I should get a Client warning that alerts me I cannot do this, and should not get a server error (or get a pretty / simple server error)

### Actual behavior
When trying to refund an amount that's greater than the balance of the order, I get a Client warning that alerts me I cannot do this, and I also get an ugly server error that should be cleaned up:

```
19:25:44.657Z ERROR Reaction: Refund amount ($25.00) is greater than unrefunded amount on charge ($21.45)
  Error: Refund amount ($25.00) is greater than unrefunded amount on charge ($21.45)
      at Constructor._Error (/Users/cassy/Documents/reaction/node_modules/stripe/lib/Error.js:12:17)
      at Constructor (/Users/cassy/Documents/reaction/node_modules/stripe/lib/utils.js:124:13)
      at new Constructor (/Users/cassy/Documents/reaction/node_modules/stripe/lib/utils.js:124:13)
      at Function.StripeError.generate (/Users/cassy/Documents/reaction/node_modules/stripe/lib/Error.js:57:14)
      at IncomingMessage.<anonymous> (/Users/cassy/Documents/reaction/node_modules/stripe/lib/StripeResource.js:170:39)
      at emitNone (events.js:111:20)
      at IncomingMessage.emit (events.js:208:7)
      at endReadableNT (_stream_readable.js:1056:12)
      at _combinedTickCallback (internal/process/next_tick.js:138:11)
      at process._tickDomainCallback (internal/process/next_tick.js:218:9)
  From previous event:
      at Constructor.create (/Users/cassy/Documents/reaction/node_modules/stripe/lib/StripeMethod.js:34:29)
      at DDPCommon.MethodInvocation.stripeRefundCreate (imports/plugins/included/payments-stripe/server/methods/stripe.js:455:44)
      at packages/check.js:128:16
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1186:15)
      at Object._failIfArgumentsAreNotAllChecked (packages/check.js:127:41)
      at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1765:18)
      at DDP._CurrentMethodInvocation.withValue (packages/ddp-server/livedata_server.js:1686:15)
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1186:15)
      at resolve (packages/ddp-server/livedata_server.js:1684:36)
      at new Promise (<anonymous>)
      at Server.applyAsync (packages/ddp-server/livedata_server.js:1683:12)
      at Server.apply (packages/ddp-server/livedata_server.js:1622:26)
      at Server.call (packages/ddp-server/livedata_server.js:1604:17)
      at DDPCommon.MethodInvocation.ordersRefundsCreate (server/methods/core/orders.js:1082:23)
      at DDPCommon.MethodInvocation._MethodHooks._wrappers.(anonymous function) (server/api/method-hooks.js:118:70)
      at packages/check.js:128:16
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1186:15)
      at Object._failIfArgumentsAreNotAllChecked (packages/check.js:127:41)
      at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1765:18)
      at DDP._CurrentMethodInvocation.withValue (packages/ddp-server/livedata_server.js:719:19)
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1186:15)
      at DDPServer._CurrentWriteFence.withValue (packages/ddp-server/livedata_server.js:717:46)
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1186:15)
      at Promise (packages/ddp-server/livedata_server.js:715:46)
      at new Promise (<anonymous>)
      at Session.method (packages/ddp-server/livedata_server.js:689:23)
      at packages/ddp-server/livedata_server.js:559:43
   => awaited here:
      at Function.Promise.await (/Users/cassy/.meteor/packages/promise/.0.10.0.10887b.zfqdkp++os+web.browser+web.cordova/npm/node_modules/meteor-promise/promise_server.js:56:12)
      at DDPCommon.MethodInvocation.stripeRefundCreate (imports/plugins/included/payments-stripe/server/methods/stripe.js:456:28)
      at packages/check.js:128:16
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1186:15)
      at Object._failIfArgumentsAreNotAllChecked (packages/check.js:127:41)
      at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1765:18)
      at DDP._CurrentMethodInvocation.withValue (packages/ddp-server/livedata_server.js:1686:15)
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1186:15)
      at resolve (packages/ddp-server/livedata_server.js:1684:36)
      at new Promise (<anonymous>)
      at Server.applyAsync (packages/ddp-server/livedata_server.js:1683:12)
      at Server.apply (packages/ddp-server/livedata_server.js:1622:26)
      at Server.call (packages/ddp-server/livedata_server.js:1604:17)
      at DDPCommon.MethodInvocation.ordersRefundsCreate (server/methods/core/orders.js:1082:23)
      at DDPCommon.MethodInvocation._MethodHooks._wrappers.(anonymous function) (server/api/method-hooks.js:118:70)
      at packages/check.js:128:16
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1186:15)
      at Object._failIfArgumentsAreNotAllChecked (packages/check.js:127:41)
      at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1765:18)
      at DDP._CurrentMethodInvocation.withValue (packages/ddp-server/livedata_server.js:719:19)
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1186:15)
      at DDPServer._CurrentWriteFence.withValue (packages/ddp-server/livedata_server.js:717:46)
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1186:15)
      at Promise (packages/ddp-server/livedata_server.js:715:46)
      at new Promise (<anonymous>)
      at Session.method (packages/ddp-server/livedata_server.js:689:23)
      at packages/ddp-server/livedata_server.js:559:43
19:25:44.658Z FATAL Reaction: Stripe call failed, refund was not issued Refund amount ($25.00) is greater than unrefunded amount on charge ($21.45)
19:25:44.658Z FATAL Reaction: Attempt for refund transaction failed sD7EZbBX7YzmXZNyA ch_1BtKumBXXkbZQs3xYLDrYDAM Cannot issue refund: Refund amount ($25.00) is greater than unrefunded amount on charge ($21.45)
```

### Steps to reproduce the behavior
- Create an order
- Capture the payment
- Attempt to refund an amount greater than the order total
- See a correct client error, but a ugly server error

### Versions

```
Node: 9.5.0
NPM: 5.6.0
Meteor Node: 8.9.3
Meteor NPM: 5.5.1
Reaction CLI: 0.27.0
Reaction: 1.8.0
Reaction branch: release-1.8.0
Docker: 18.01.0-ce
```

### QA

QA by @kieckhafer + @cassytaylor

**Comment by aldeed** [comment]
@kieckhafer Another one I suspect you may have resolved

**Comment by kieckhafer** [comment]
Yes, #5158 will take care of this.

**Comment by kieckhafer** [comment]
Closed by #5158
